### PR TITLE
feat(codedeploy): termination hook

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

None

### Reason for this change

Deployment group supports a termination hook for EC2 compute type but CDK L2 construct does not support this.

### Description of changes

Added `terminationHook` to `ServerDeploymentGroupProps`

```ts
    new codedeploy.ServerDeploymentGroup(stack, 'DeploymentGroup', {
      autoScalingGroups: [
        new autoscaling.AutoScalingGroup(stack, 'ASG', {
          instanceType: ec2.InstanceType.of(ec2.InstanceClass.STANDARD3, ec2.InstanceSize.SMALL),
          machineImage: new ec2.AmazonLinuxImage(),
          vpc: new ec2.Vpc(stack, 'VPC'),
        }),
      ],
      terminationHook: true, // add
    });
```

### Description of how you validated changes

Added both unit and integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts [hunk 1]
```diff
@@ -40,6 +41,9 @@ new codedeploy.ServerDeploymentGroup(stack, 'CodeDeployGroup', {
     failedDeployment: false,
     deploymentInAlarm: false,
   },
+  terminationHook: true,
 });
 
-app.synth();
+new IntegTest(app, 'integ-cloudfront-custom-s3', {
+  testCases: [stack],
+});
```
Comment (msambol) [inline packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts:49]: nit: can you add newline?

Comment (badmintoncryer) [reply packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts]: Thanks! I've added!

File: packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts [hunk 2]
```diff
@@ -40,6 +41,9 @@ new codedeploy.ServerDeploymentGroup(stack, 'CodeDeployGroup', {
     failedDeployment: false,
     deploymentInAlarm: false,
   },
+  terminationHook: true,
 });
 
-app.synth();
+new IntegTest(app, 'integ-cloudfront-custom-s3', {
```
Comment (GavinZZ) [inline packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts:47]: I might miss something but the test suite name seems weird to me.

**Review by GavinZZ — COMMENTED** [review]
Look good in general, just a nit question on the test name.

File: packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts (hunk 2 above)
Comment (badmintoncryer) [reply packages/@aws-cdk-testing/framework-integ/test/aws-codedeploy/test/server/integ.deployment-group.ts]: I'm sorry...　I've updated test name.

**Comment by badmintoncryer** [comment]
@GavinZZ Thank you for your review! I've addressed your comments.

**Comment by badmintoncryer** [comment]
I'll update snapshot files later

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**